.github/workflows/buf-ci.yaml

```diff
@@ -0,0 +1,29 @@
+name: Buf CI
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-alpha.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+"
+  pull_request:
+permissions:
+  contents: read
+  pull-requests: write
+jobs:
+  buf:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: bufbuild/buf-action@v1
+        with:
+          token: ${{ secrets.BUF_TOKEN }}
+          input: "pb"
+          breaking-against: 'https://github.com/celestiaorg/nmt.git#branch=main,ref=HEAD~1,subdir=pb'
+          # Run breaking change, lint, and format checks for Protobuf sources against all branches,
+          # 'pb' # subdirectory, then push to the BSR once validated
```

Comment:
This line doesn't make sense to me. 

1. Why two `#`?
2. What is 'pb'?
3. Can this use the longer format Buf Schema Registry for the first usage of BSR. So something like "then push to the Buf Schema Registry (BSR) once validated

Reply:
1. [762e67b](https://github.com/celestiaorg/nmt/pull/274/commits/762e67bacd69a96fbaf757bf1034014109ff476b)
2. folder containing the protos
3. [3fb737f](https://github.com/celestiaorg/nmt/pull/274/commits/3fb737f7cb7c4c087743684955a2c48e0b767734)